api/apps/api/src/modules/published-project/published-project-crud.service.ts

```diff
@@ -73,6 +77,24 @@ export class PublishedProjectCrudService extends AppBaseService<
     };
   }
 
+  async extendGetByIdResult(
+    entity: PublishedProject,
+    _fetchSpecification?: FetchSpecification,
+    _info?: ProjectsRequest,
+  ): Promise<PublishedProject> {
+    const exportIdString = entity?.exportId;
+
+    if (exportIdString) {
+      const exportId = new ExportId(exportIdString);
+      const finalExport = await this.exportRepo.find(exportId);
+      if (!finalExport) {
+        delete entity.exportId;
+      }
+    }
```

Comment:
If published project has an export id it should always exist on export repository (at least that we are doing something strange when publishing projects). If I remember correctly, Miguel wanted export id property only if the export has finished right? We should check if the export has finished after retrieving it from export repo. We usually check that by inspecting `archiveLocation` property inside `Export` aggregate. I think it is worth to create a method `isFinished(): boolean` to check that condition internally 🤔, I can take care if you want 👍  

```suggestion
    const exportIdString = entity?.exportId;

    if (exportIdString) {
      const exportId = new ExportId(exportIdString);
      const finalExport = await this.exportRepo.find(exportId);
      if (!finalExport?.isFinished()) {
        delete entity.exportId;
      }
    }
```